In [7]:
%load_ext blackcellmagic

In [1]:
import cv2
import glob
import sys

In [2]:
filenames = glob.glob("../DataSets/Cats_and_Dogs/*")

In [7]:
def generate_basic_metadata(filenames):
    
    num_files = len(filenames)
    image_metadata = {}
    
    for i, filename in enumerate(filenames):

        image = cv2.imread(filename)
        height, width, channels = image.shape
        std, mean, min, max, sum = (
            image.std(),
            image.mean(),
            image.min(),
            image.max(),
            image.sum(),
        )

        image_metadata[filename] = {
            "height": height,
            "width": width,
            "channels": channels,
            "std": std,
            "mean": mean,
            "min": min,
            "max": max,
            "sum": sum,
        }
        
        sys.stdout.write("\r{:.2f}%".format(100*(i + 1) / num_files))
    sys.stdout.write("\r\n")
    return image_metadata


In [8]:
%%time
metadata = generate_basic_metadata(filenames)

100.00%CPU times: user 6min 45s, sys: 23.9 s, total: 7min 9s
Wall time: 3min 40s


MacBook

CPU times: user 6min 45s, sys: 23.9 s, total: 7min 9s
Wall time: 3min 40s